src/parachain/vaults.ts

```diff
@@ -588,20 +587,27 @@ export class DefaultVaultsAPI extends DefaultTransactionAPI implements VaultsAPI
     }
 
     async getTotalIssuableAmount(): Promise<MonetaryAmount<WrappedCurrency, BitcoinUnit>> {
-        const perCollateralIssuableAmounts = await Promise.all(
-            CollateralCurrency.map(
-                (currency) =>
-                    new Promise<MonetaryAmount<WrappedCurrency, BitcoinUnit>>((resolve) => {
-                        this.tokensAPI.total(currency as Currency<CurrencyUnit>).then(async (nomination) => {
-                            const capacity = await this.calculateCapacity(
-                                nomination as MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>
-                            );
-                            resolve(capacity);
-                        });
-                    })
-            )
-        );
-        const totalIssuableAmount = perCollateralIssuableAmounts.reduce((acc, v) => acc.add(v));
+        // get [[wrapped, collateral], amount][] map
+        const perCurrencyPairCollateralAmounts = await this.api.query.vaultRegistry.totalUserVaultCollateral.entries();
+        // filter for wrapped === this.wrapped (as only one wrapped currency is handled at a time currently?)
+        const perWrappedCurrencyCollateralAmounts = perCurrencyPairCollateralAmounts.filter(([key, _val]) =>
+            currencyIdToMonetaryCurrency(key.args[0].wrapped).name === this.wrappedCurrency.name);
+        // reduce from [[this.wrapped, collateral], amount][] pairs to [collateral, sumAmount][] map
+        const perCollateralCurrencyCollateralAmounts = perWrappedCurrencyCollateralAmounts.reduce((amounts, [key, amount]) => {
+            const collateralCurrency = currencyIdToMonetaryCurrency<CollateralUnit>(key.args[0].collateral);
+            let collateralAmount = newMonetaryAmount(amount.toString(), collateralCurrency);
+            if (amounts.has(collateralCurrency)) {
+                // .has() is true, hence non-null
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                collateralAmount = collateralAmount.add(amounts.get(collateralCurrency)!);
+            }
+            amounts.set(collateralCurrency, collateralAmount);
+            return amounts;
+        }, new Map<Currency<CollateralUnit>, MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>>());
+        // finally convert the CollateralAmount sums to issuable amounts and sum those to get the total
+        const perCollateralCurrencyIssuableAmounts = await Promise.all(
+            [...perCollateralCurrencyCollateralAmounts.values()].map((collateralAmount) => this.calculateCapacity(collateralAmount)));
```

Comment:
It's great that you preprocess `perWrappedCurrencyCollateralAmounts` into `[wrapped, collateral]` pairs before calling `calculateCapacity`. Otherwise there would have been unnecessary queries for the exchange rate and secure threshold